wire-ios-data-model/Source/Utilis/EAR/EARKeyEncryptor.swift

```
@@ -0,0 +1,77 @@
+//
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+// sourcery: AutoMockable
+protocol EARKeyEncryptorInterface {
+
+    func encryptDatabaseKey(
+        _ databaseKey: Data,
+        publicKey: SecKey
+    ) throws -> Data
+
+    func decryptDatabaseKey(
+        _ encryptedDatabaseKey: Data,
+        privateKey: SecKey
+    ) throws -> Data
+
+}
+
+final class EARKeyEncryptor: EARKeyEncryptorInterface {
+
+    private var databaseKeyAlgorithm: SecKeyAlgorithm {
+        return .eciesEncryptionCofactorX963SHA256AESGCM
+    }
+
+    func encryptDatabaseKey(
+        _ databaseKey: Data,
+        publicKey: SecKey
+    ) throws -> Data {
+        var error: Unmanaged<CFError>?
+        guard let encryptedDatabaseKey = SecKeyCreateEncryptedData(
+            publicKey,
+            databaseKeyAlgorithm,
+            databaseKey as CFData, &error
+        ) else {
+            let error = error!.takeRetainedValue() as Error
```

safe to force unwrap?

---

I guess so, but we could also just throw an "unknown" error if we can't unwrap it.